### PR TITLE
Aggregate analytics event data

### PR DESCRIPTION
**Pull request by aldenstpage**
## Description
We have a large event stream that tracks assorted interesting usage data of CC Search. We want to build a dashboard that shows daily summaries of search. To do this, we have to aggregate the data on a daily basis. This PR adds some scripts and tables for producing this aggregated data, including:

- Usage reports with general statistics about detail page events, number of unique sessions, number of searches, attribution hits, search ratings, and searches per session
- Summaries of how many of each source received a click
- Summaries of domains with embedded CC Search attribution data and how many hits each one receives
- Summaries of top searches
- Scoreboard for tracking which images received the most clicks

## Tests
See additional unit tests

### CODE REVIEW

**Comment by zackkrida** [comment]
@aldenstpage Initial impressions are good. The migration tables and columns make sense, and the logic in the reporting controller checks out. I'll be able to review a little more thoroughly in a few hours, if you need this approved earlier let me know.

**Comment by aldenstpage** [comment]
Take your time!

File: analytics/report_controller.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        DetailPageEvent.timestamp < end_time,
+        DetailPageEvent.event_type == DetailPageEvents.SHARED_SOCIAL
+    ).count()
+    sessions = session.query(
+        func.count(
+            distinct(SearchEvent.session_uuid)
+        ).filter(
+            SearchEvent.timestamp > start_time,
+            SearchEvent.timestamp < end_time
+        )
```
Comment (zackkrida) [inline analytics/report_controller.py:48]: Just a reminder that `session_uid` is currently cached for 7 days for each user and I'll let you know when it's fixed.

**Review by zackkrida — APPROVED** [review]
This is great. Nice work

**Pull request merged**